spec/features/collection_multi_membership_spec.rb

```diff
@@ -0,0 +1,185 @@
+RSpec.describe 'Adding a work to multiple collections', type: :feature, clean_repo: true, js: true, with_nested_reindexing: true do
+  let!(:admin_user) { create(:admin, email: 'admin@example.com') }
+  let!(:single_membership_type_1) { create(:collection_type, :not_allow_multiple_membership, title: 'Single-membership 1') }
+  let!(:single_membership_type_2) { create(:collection_type, :not_allow_multiple_membership, title: 'Single-membership 2') }
+  let!(:multi_membership_type_1) { create(:collection_type, :allow_multiple_membership, title: 'Multi-membership 1') }
+  let!(:multi_membership_type_2) { create(:collection_type, :allow_multiple_membership, title: 'Multi-membership 2') }
+
+  let!(:col1_sm_1) { create(:collection_lw, id: 'col1_sm_1', title: ['Collection 1 for SM1'], user: admin_user, collection_type_gid: single_membership_type_1.gid) }
+  let!(:col2_sm_1) { create(:collection_lw, id: 'col2_sm_1', title: ['Collection 2 for SM1'], user: admin_user, collection_type_gid: single_membership_type_1.gid) }
+  let!(:col3_sm_2) { create(:collection_lw, id: 'col3_sm_2', title: ['Collection 3 for SM2'], user: admin_user, collection_type_gid: single_membership_type_2.gid) }
+  let!(:col4_mm_1) { create(:collection_lw, id: 'col4_mm_1', title: ['Collection 4 for MM1'], user: admin_user, collection_type_gid: multi_membership_type_1.gid) }
+  let!(:col5_mm_1) { create(:collection_lw, id: 'col5_mm_1', title: ['Collection 5 for MM1'], user: admin_user, collection_type_gid: multi_membership_type_1.gid) }
+  let!(:col6_mm_2) { create(:collection_lw, id: 'col6_mm_2', title: ['Collection 6 for MM2'], user: admin_user, collection_type_gid: multi_membership_type_2.gid) }
+
+  let!(:work) { create(:generic_work, user: admin_user, title: ['The highly valued work that everyone wants in their collection']) }
+
+  before do
+    admin_user
```

Comment:
No need to call this - you're using `let!` so the object is already created (not lazy) - and even if you were using the lazy `let` it would be created in the next line when you call it again.